src/navigation/navigation-events.js

```diff
@@ -148,49 +141,7 @@ function createPopStateEvent(state) {
 window.singleSpaNavigate = navigateToUrl;
 
 function parseUri(str) {
-  // parseUri 1.2.2
-  // (c) Steven Levithan <stevenlevithan.com>
-  // MIT License
-  // http://blog.stevenlevithan.com/archives/parseuri
-  const parseOptions = {
-    strictMode: true,
-    key: [
-      "source",
-      "protocol",
-      "authority",
-      "userInfo",
-      "user",
-      "password",
-      "host",
-      "port",
-      "relative",
-      "path",
-      "directory",
-      "file",
-      "query",
-      "anchor"
-    ],
-    q: {
-      name: "queryKey",
-      parser: /(?:^|&)([^&=]*)=?([^&]*)/g
-    },
-    parser: {
-      strict: /^(?:([^:\/?#]+):)?(?:\/\/((?:(([^:@]*)(?::([^:@]*))?)?@)?([^:\/?#]*)(?::(\d*))?))?((((?:[^?#\/]*\/)*)([^?#]*))(?:\?([^#]*))?(?:#(.*))?)/,
-      loose: /^(?:(?![^:@]+:[^:@\/]*@)([^:\/?#.]+):)?(?:\/\/)?((?:(([^:@]*)(?::([^:@]*))?)?@)?([^:\/?#]*)(?::(\d*))?)(((\/(?:[^?#](?![^?#\/]*\.[^?#\/.]+(?:[?#]|$)))*\/?)?([^?#\/]*))(?:\?([^#]*))?(?:#(.*))?)/
-    }
-  };
-
-  let o = parseOptions;
-  let m = o.parser[o.strictMode ? "strict" : "loose"].exec(str);
-  let uri = {};
-  let i = 14;
-
-  while (i--) uri[o.key[i]] = m[i] || "";
-
-  uri[o.q.name] = {};
-  uri[o.key[12]].replace(o.q.parser, function($0, $1, $2) {
-    if ($1) uri[o.q.name][$1] = $2;
-  });
-
-  return uri;
+  const anchor = document.createElement("a");
+  anchor.href = str;
+  return anchor;
```

Comment:
wow this is quite the update

is an anchor tag the same return object shape as what was previously returned from this function?

Reply:
Not quite - `path` becomes `pathname`, etc. This accounts for the changes I made later on in the navigateToUrl function